articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,147 @@
+
+![image](./server1.png)  
+
+**Client app running on Port 3001**  
+
+![image](./client1.png)
+
+Now that we have our apps up and running, we don't want our users to use these applications by typing their PORTS explicitly, so we need to map it with something which is more human-readable. In this example, I will be using subdomains to distinguish between them. Again one is free to use whichever distinguishing element suitable as per their requirement.
+
+## Step 2: Add DNS records
+This is the part where one would add the DNS records in their DNS management dashboard. Since I am running it all locally I wouldn't be needing that.  
+
+The general DNS Configurations would be something like
+- Server app mapped to the server.domain
+- Client app mapped to the client.domain
+
+My Localhost Config, in this case, would be
+- Server mapped to server.localhost
+- Client mapped to client.localhost
+
+There are two standard protocols HTTP and HTTPS. The default port for HTTP is 80 and HTTPS is 443. This is the reason we must not run our applications on these ports because our NGINX server would be running on these two ports. All the requests the client makes would either be redirected to Port 80 or 443 from where it would be redirected internally to the corresponding application.
+
+## Step 4 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX to make it all work.
+Navigate into the Nginx folder and locate  ```nginx.conf``` .
+
+Add these configurations inside the HTTP block.
+
+### Step 4.1 - HTTP
+
+```
+server {       
+     listen       80;
+     server_name  server.domain;
+     location / {
+          proxy_pass "http://localhost:3000" ;
+     }
+}
+server {       
+     listen       80;
+     server_name  client.domain;
+
+     location / {
+          proxy_pass "http://localhost:3001" ;
+     }
+}
+```
+
+### Step 4.2 - HTTPS
+
+```
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name server.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3000/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+server{
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name client.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3001/";
+    }
+
+    ssl_certificate <location of SSL certificate>
+    ssl_certificate_key <location of SSL certificate Key>
+}
+```
+
+**NOTE** 
+These are the basic minimum configurations required to successfully configure NGINX for reverse proxying. Feel free to explore other config parameters as well. 
+
+Change the domain name to your domain.
+For SSL Certificate and Key, you can get them from your SSL provider. If you don't have one, Use this free service [LetsEncrypt](https://letsencrypt.org/). Follow their documentation to get Free SSL instantly!
```

Comment:
For **a** Replace to get with to obtain for flow.

Reply:
I don't find any **a** here, I must be missing something @louisefindlay23

Reply:
Sorry, wasn't very clear @HarishTeens. For **a** SSL certificate and key, you can **obtain them**.

Add **a **to make the sentence make sense.

**obtain them** is for flow/readability.

Reply:
okay gotcha 👍

Reply:
pushed commits for these as well